File: output/CodeGen/codenet/C++/Python/s166776347.py
def main():
    # 入力
    N,K = map(int,raw_input().split())
    A = map(int,raw_input().split())

    # 解法 
    ans1 = 0
    for i in range(N):
        for j in range(N):
            if i >= j: continue
            if A[i] > A[j]: ans1 += 1
            ans1 %= mod

    # 解法2
    count = [0]*2005
    for i in range(N):
        count[A[i]-1] += 1
    remain = N
    for i in range(2005):
        remain -= count[i]
        ans2 += count[i]*remain
    K %= mod
    ans2 %= mod
    ans1 *= K
    ans1 %= mod
    ans2 *= K
    ans2 %= mod
    ans = ans1 + ans2
    ans %= mod
    ans += K*(K-1)/2
    ans %= mod
    print ans


if __name__ == '__main__':
    main()

File: output/CodeGen/codenet/C++/Java/s166776347.java
public class s166776347 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int k = in.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = in.nextInt();
        }
        int ans1 = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i > j) continue;
                if (a[i] > a[j]) ans1++;
                ans1 %= 1000000007;
            }
        }
        int[] count = new int[2005];
        for (int i = 0; i < n; i++) {
            count[a[i] - 1]++;
        }
        int ans2 = 0;
        int remain = n;
        for (int i = 0; i < 2005; i++) {
            remain -= count[i];
            ans2 += count[i] * remain;
        }
        int ans = 0;
        int tmp = k * (k - 1) / 2;
        tmp %= 1000000007;
        ans += ans1 * k;
        ans %= 1000000007;
        ans += tmp;
        ans %= 1000000007;
        System.out.println(ans);
    }
}